datascience/predicates.py

```diff
@@ -237,5 +248,13 @@ def _equal_or_float_equal(x, y):
     else:
         return x == y
 
+############
+# Utility #
+############
+def check_iterable(y):
+    if hasattr(y, '__iter__'):
+            warnings.warn("You are checking against an iterable. If you are trying to test one iterable against another \
+                using 'Table.where', put your first iterable as your first argument, your predicate as your second  \
+                argument, and then your second iterable as your third argument.")
```

Comment:
Let's see if we can improve the warning.  I'm not sure if this will mean anything to students ("iterable", "checking against an iterable".)  How about "Do not pass an array or list to a predicate.  If you are trying to find rows where two columns are the same, use table.where('c', are.equal_to, 'd') instead of table.where('c', are.equal_to(table.column('d')))."

Reply:
Agreed! I've made the change